File: Section Reports/Layout/SubReport/C#/SubReport/rptParamMain.cs
using System;
namespace GrapeCity.ActiveReports.Samples.SubReport
{
	/// <summary>
	/// Description of the outline of the rptParamMain
	/// 
	/// </summary>
	public partial class rptParamMain : GrapeCity.ActiveReports.SectionReport
	{
		rptParamSub _rpt;
		public rptParamMain()
		{
			//
			// ActiveReport designer support is required
			//
			//
			InitializeComponent();
		}
		private void Detail_Format(object sender, EventArgs e)
		{
			SubReport1.Report = _rpt;
		}
		private void rptParamMain_ReportStart(object sender, EventArgs e)
		{
			_rpt = new rptParamSub();
			Data.OleDBDataSource dS = new Data.OleDBDataSource();
			dS.ConnectionString = Properties.Resources.ConnectionString;			
			dS.SQL = "Select * from suppliers order by country";
			DataSource = dS;
			
			reportInfo1.FormatString = "Page {PageNumber} of {PageCount}";
			
			
		}
	}
}

File: Section Reports/Layout/SubReport/VB.NET/SubReport/rptParamMain.vb
Public Class rptParamMain
	Inherits GrapeCity.ActiveReports.SectionReport
	Dim rpt As rptParamSub
	Public Sub New()
		MyBase.New()
		InitializeComponent()
	End Sub
	Private Sub Detail_Format(ByVal sender As Object, ByVal e As EventArgs) Handles Detail.Format
		SubReport1.Report = rpt
	End Sub
	
	Private Sub rptParamMain_ReportStart(sender As System.Object, e As System.EventArgs) Handles MyBase.ReportStart
		rpt = Nothing
		rpt = New rptParamSub()
		Dim dS As New Data.OleDBDataSource()
		dS.ConnectionString = My.Resources.ConnectionString
		dS.SQL = "Select * from suppliers order by country"
		DataSource = dS
		
		ReportInfo1.FormatString = "Page {PageNumber} of {PageCount}"
		
		'
	End Sub
End Class
